### PR TITLE
update guide for Gradle 5/5.1

### PR DESCRIPTION
**Pull request by jlstrater**
- add kotlin dsl and groovy dsl examples
- update init plugin output
- update test summary to reflect package name change

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,102 +1,130 @@
...
-    mainAuthor 'Schalk Cronjé'
+    repoPath 'gradle-guides/building-groovy-libraries'
 }
 
 asciidoctor {
-    attributes 'gradle-outdir' : "${gradleRunner.reportsDir.absolutePath}"
+    attributes 'gradle-outdir': "${gradleRunner.reportsDir.absolutePath}"
 }
 
 gradleRunner {
```
Comment (eriwen) [inline build.gradle]: :o: It'd be cool to eliminate use of this plugin and switch to Exemplar, but we shouldn't block on that. 

You can find a good example in the https://github.com/gradle-guides/implementing-gradle-plugins/ repo. 

See also https://github.com/gradle-guides/implementing-gradle-plugins/blob/master/build.gradle.kts#L9-L18

File: contents/index.adoc [hunk 2]
```diff
@@ -26,61 +26,81 @@ $ cd building-groovy-libraries
 
 From inside the new project directory, run the `init` task with the `groovy-library` argument.
 
-[listing]
+.If you prefer the Groovy DSL
```
Comment (eriwen) [inline contents/index.adoc]: :x: We have much nicer and standardized ways to handle multiple DSLs: consider [this example](https://github.com/gradle-guides/implementing-gradle-plugins/blame/master/contents/index.adoc#L52) using the `include::sample` macro.

File: contents/index.adoc [hunk 3]
```diff
@@ -26,61 +26,81 @@ $ cd building-groovy-libraries
...
 <3> Default Groovy test folder
 
 == Review the generated project files
 
-The `settings.gradle` file is heavily commented, but has only one active line:
+The `settings.gradle`(or `settings.gradle.kts`) file is heavily commented, but has only one active line:
 
-.Generated settings.gradle
-[source,groovy]
+[source.multi-language-sample,groovy]
```
Comment (eriwen) [inline contents/index.adoc]: Consider `include::sample` macro here too, and in other places.

File: build.gradle [hunk 4]
```diff
@@ -1,102 +1,130 @@
...
-    step 'editLibrary', {
-        File editLibrary = new File(workingDir,'src/main/groovy/Library.groovy')
-        editLibrary.text = "${file('contents/groovydoc-prefix.txt').text}${editLibrary.text}"
-    }
+        step "${sample.name}-addTags", {
+            File buildPrefix = project.file("contents/${sample.name}/build-prefix.txt")
+            File editBuildFile = new File(workingDir, "build${sample.scriptExtension}")
+            String editBuildFileText = editBuildFile.text
+            File buildPostfix = project.file("contents/${sample.name}/build-postfix.txt")
+            editBuildFile.withWriter { Writer writer ->
```
Comment (eriwen) [inline build.gradle]: I don't think we'll need all this when using the samples infrastructure we have setup.

**Comment by jlstrater** [comment]
Ok @eriwen. I copied this pattern from the @gradle-guides/building-java-libraries example. Should we go back and update that one too?

**Comment by eriwen** [comment]
@jlstrater Yeah, it's up to us when to do that. I won't block this change though if you want to be done for now. There's an issue for Exemplar just like the ones for 5.0: https://github.com/gradle/guides/issues/192

**Comment by lacasseio** [comment]
All possibly great change for the guides, but it was left unmerge for so long. If you still feel strongly about the change please reopen another PR.

**Pull request closed without merging**